src/Build/Instance/ProjectInstance.cs

```diff
@@ -2082,7 +2082,12 @@ internal static ProjectInstance[] LoadSolutionForBuild(string projectFile, Prope
                         toolsVersion = visualStudioVersion.ToString(CultureInfo.InvariantCulture) + ".0";
                     }
 
-                    string toolsVersionToUse = Utilities.GenerateToolsVersionToUse(explicitToolsVersion: null, toolsVersionFromProject: toolsVersion, getToolset: buildParameters.GetToolset, defaultToolsVersion: Constants.defaultSolutionWrapperProjectToolsVersion);
+                    string toolsVersionToUse = Utilities.GenerateToolsVersionToUse(
+                        explicitToolsVersion: null,
```

Comment:
`explicitToolsVersion` is a horrible name and threw me off initially. It does not refer to the toolsversion being explicit in the file, but to the toolsversion being specified via the APIs. I suggest renaming it everywhere to to `explicitToolsVersionViaApi`. Sounds good?